.github/workflows/CI.yaml

```diff
@@ -151,15 +151,60 @@ jobs:
         with:
           python-version: 3.9
 
-      - name: Install dependencies
+      - name: Upgrade pip and install tomli
         run: |
           python -m pip install --upgrade pip
           python -m pip install tomli
-          eval "python -m pip install $(./bin/min_dependencies.py)"
-          python -m pip install -e ".[test]"
+      - name: Install minimum dependencies
+        run: eval "python -m pip install $(./bin/min_dependencies.py)"
+      - name: Install anesthetic
+        run: python -m pip install -e ".[test]"
 
       - name: Test with pytest
         run: python -m pytest --cov=anesthetic tests
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1
```

Comment:
Should we remove the codecov upload for `minimum-dependencies`, too? Not really what this job is about either, right?

Reply:
I disagree, since it is necessary that anesthetic works with the lowest versions we allow, and a PR should not be merged unless this is true

Reply:
But codecov has nothing to do with whether anesthetic works or not, that is what pytest is for. Codecov should definitely be in the pip and conda jobs (or at least in one of them), but the coverage does not really depend on versions (unless we specifically add version related conditional statements to the code, which we don't). Besides, the `minimum-dependencies` job does not even include extra dependencies, so it doesn't do a full code coverage anyhow...

Reply:
_If_ the `minimum-dependencies` code coverage were to fail, while the pip code coverage passes, then we wouldn't even know about it, because as long as the pip jobs cover every line at least once, codecov will mark 100% coverage. But I don't even think it is possible for code coverage to be different for the pip jobs and the minimum job...

Reply:
Some of the tests are now different for matplotlib≥3.8, and at one point in beta some of the anesthetic plotting code was different for pandas 1 vs 2. cov would tell us when these lines are no longer needed right?

Reply:
> Some of the tests are now different for matplotlib≥3.8, 

Code in `tests/` doesn't matter for code coverage, only code in `anesthetic/` matters.

> and at one point in beta some of the anesthetic plotting code was different for pandas 1 vs 2. cov would tell us when these lines are no longer needed right?

In a situation like that we would indeed need the code coverage in `minimum-dependencies` for as long as we wanted to be backwards compatible. However, we already have pandas>=2.0.0, so this shouldn't be an issue, here.

In fact, I think this might be an argument in favour of not having code coverage in `minimum-dependencies`, to flag up once parts of code are no longer needed for newer versions. Currently we wouldn't know if parts of code were no longer needed by newer versions.

Reply:
> > and at one point in beta some of the anesthetic plotting code was different for pandas 1 vs 2. cov would tell us when these lines are no longer needed right?
> 
> In a situation like that we would indeed need the code coverage in `minimum-dependencies` for as long as we wanted to be backwards compatible. However, we already have pandas>=2.0.0, so this shouldn't be an issue, here.
> 
> In fact, I think this might be an argument in favour of not having code coverage in `minimum-dependencies`, to flag up once parts of code are no longer needed for newer versions. Currently we wouldn't know if parts of code were no longer needed by newer versions.

I guess it depends how cut-throat we want to be with older packages. I suppose the pandas2 difference was because pandas2 hadn't actually released yet, and once it did, we dropped support for pandas1.

Reply:
> > Some of the tests are now different for matplotlib≥3.8,
> 
> Code in `tests/` doesn't matter for code coverage, only code in `anesthetic/` matters.

Ah - I'd thought `pytest --cov anesthetic tests` meant that coverage was looking at anesthetic AND tests, but tests is just to tell pytest where the tests are. I don't think I specified this in the new pytest calls

Reply:
I agree that there might be a point where we'll want to re-introduce this for backwards compatibility, but I'd rather be aware when that happens.